lib/view_model/deserialization_error.rb

```diff
@@ -0,0 +1,55 @@
+class ViewModel
+  class DeserializationError < StandardError
+    attr_reader :nodes
+    def initialize(msg, nodes = [])
+      super(msg)
+      @nodes = Array.wrap(nodes)
+    end
+
+    def http_status
+      400
+    end
+
+    def metadata
```

Comment:
Should this include the exception type? Is that communicated to the client in any way other than the http status? I can't see here how the client would distinguish "stale update", which it should definitely handle internally, vs "request format was completely garbage", which is a API consumer error and can't reasonably be expected to be handled.